### ISSUE TITLE
v0.70.1: Live build_minillm_callback (teacher-mix sampling + length-norm + pretrain-loss anchor)

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#237]
v0.70.0 Part C ships the MiniLLM schema (`MiniLLMConfig` frozen, anchor weight↔path mutual-requirement cross-validators) + 5 TrainingConfig fields. The deferred-live stub raises `NotImplementedError` with v0.70.1 marker.

What's needed in v0.70.1:

1. Wire `build_minillm_callback(config: MiniLLMConfig)` to return an HF Trainer callback that:
   - At rollout time, with probability `teacher_mix_ratio`, samples from teacher logits instead of student (epsilon-greedy mix).
   - Length-normalises the rollout log-probability before the reverse-KL term when `length_normalize=True`.
   - When `pretrain_anchor_weight > 0`, adds the SFT-on-pretrain loss term scaled by `pretrain_anchor_weight` to the total loss.
2. Apply cwd-containment + symlink rejection on `pretrain_anchor_path` at the disk-read site (deferred at schema time per the v0.69.0 build_dag `source` policy).
3. Integrate with v0.53.2 DistillTrainerWrapper so MiniLLM + ULD can compose.
4. Acceptance criteria:
   - Unit test: callback fires on `on_step_end` and modifies loss.
   - End-to-end smoke: Llama-3.2-1B distillation runs 10 steps with `--minillm-enabled --minillm-teacher-mix-ratio 0.3` without crashing.
5. Schema is in place; v0.70.1 wires the callback only.

**Comment by MakazhanAlpamys** [comment]
Shipped in v0.71.11. See [release notes](https://github.com/MakazhanAlpamys/Soup/releases/tag/v0.71.11).

Try: `soup train --config distill.yaml --minillm-enabled --minillm-teacher-mix-ratio 0.3 --minillm-pretrain-anchor-path ./pretrain.jsonl --minillm-pretrain-anchor-weight 0.1`
